src/lib/utilities/Dialog/stores.ts

```diff
@@ -2,30 +2,38 @@
 
 import { writable } from 'svelte/store';
 
-export interface DialogAlert {
-	icon?: string;
-	title: string;
-	body: string;
-	image?: any;
-	component?: any;
-	html?: any;
-}
-export interface DialogConfirm extends DialogAlert {
-	type: string;
-	result: any;
+export interface DialogComponent {
+	/** Import and provide your component reference. */
+	ref: any;
+	/** Provide a key/value pairs of component props. */
+	props?: Record<string, string>;
+	/** Provide an HTML template literal for the default slot. */
+	slot?: string;
 }
-export interface DialogPrompt extends DialogAlert {
-	type: string;
-	value: any;
-	result: any;
+
+export interface Dialog {
+	/** Designate what type of component will display. */
+	type: 'alert' | 'confirm' | 'prompt' | 'component';
+	/** Provide the dialog header content. Accepts HTML. */
+	title?: string;
+	/** Provide the dialog body content. Accepts HTML. */
+	body?: string;
+	/** Provide a URl to display an image within the dialog. */
+	image?: any;
+	/** Prompt Only - provide a prompt value. */
+	value?: any;
+	/** Provide {component, props} to create a generate a custom component. */
+	component?: DialogComponent;
+	/** Provide a function. Returns the repsonse value. */
+	response?: any;
```

Comment:
unknown

Reply:
@ryceg This should always be a function. is there a better type for that? I realize most functions are Objects, like everything in JS, but that doesn't seem right.

Reply:
What type does the `response` value return? You can mark it as a function like `response?(): unknown;`, where the `unknown` is what is returned.

Reply:
It any be anything, we know what our canned options will provide, but custom dialogs use this as well and can return anything.

Reply:
Does the callback return anything? It looks like it never is, so in that case, you'd be looking at something like: `response?: (arg: unknown) => void;`